queue_processors/autoreduction_processor/post_process_admin.py

```diff
@@ -208,16 +208,29 @@ def reduce(self):
             script_out = os.path.join(log_dir, log_and_err_name + "Script.out")
             mantid_log = os.path.join(log_dir, log_and_err_name + "Mantid.log")
 
-            # strip the temp path off the front of the temp directory to get the final archives
-            # directory.
-            final_result_dir = reduce_result_dir[len(MISC["temp_root_directory"]):]
-            final_log_dir = log_dir[len(MISC["temp_root_directory"]):]
-
-            final_result_dir = self._new_reduction_data_path(final_result_dir)
-            final_log_dir = append_path(final_result_dir, ['reduction_log'])
-
-            logger.info('Final Result Directory = %s', final_result_dir)
-            logger.info('Final Log Directory = %s', final_log_dir)
+            def result_directory():
+                """
+                create and return final result directory, stripping temporary path off the front
+                of temporary directories
+                :return (str) final result directory path
+                """
+                final_result_directory = reduce_result_dir[len(MISC["temp_root_directory"]):]
```

Comment:
Shame were not up to Python 3.9 else we could have done `reduce_result_dir.removeprefix(MISC["temp_root_directory"])`

Could we do either an assert (if we know this always should be true) or an if for that:
`reduce_result_dir.starts_with(MISC["temp_root_directory"])` before slicing

Reply:
Also could the caller pass in these fields at the point of call? I know they're global, but this slowly fixes the global usage and makes it trivial to unit test as a free function

Something like:
`def result_directory(reduce_result_dir, root_directory_dict)` ?